src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -55,10 +55,7 @@ def parse_timestamp(ts: str) -> datetime:
 def parse_retention(retention: str) -> Dict[str, int]:
     ret = {}
     log.debug(f'parse_retention({retention})')
-    matches = re.findall(r'\d+[a-z]', retention)
-    for m in matches:
-        ret[m[-1]] = int(m[0:-1])
-    matches = re.findall(r'\d+[A-Z]', retention)
+    matches = re.findall(r'\d+[nmhdwMy]', retention)
```

Comment:
What's the connection used is mgr/snap_schedule for retention - upper or lower case? I remember it got changes some time ago, so just double checking in case we need to readjust :)

Reply:
mostly lowercase ... but synced with other subsystems i.e. rbd, rgw
the only caveat is (m)inutes and (M)onths in case of CephFS
the idea for this PR is to use only the required subset characters for match and not a superset of all uppercase and lowercase chars